interfaces/builtin/iotedge.go

```diff
@@ -0,0 +1,55 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2020 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package builtin
+
+const iotedgeSummary = `allows Azure iotedge daemon and client`
+
+const iotedgeBaseDeclarationSlots = `
```

Comment:
I think this needs to be changed, something like what we have for docker-support:

```go
const dockerSupportBaseDeclarationPlugs = `
  iotedge:
    allow-installation: false
    deny-auto-connection: true
`

const iotedgeBaseDeclarationSlots = `
  iotedge:
    allow-installation:
      slot-snap-type:
        - core
    deny-auto-connection: true
`


```